app_test.go

```diff
@@ -1422,7 +1423,7 @@ func TestApp_Run_CommandWithSubcommandHasHelpTopic(t *testing.T) {
 	}
 
 	for _, flagSet := range subcommandHelpTopics {
-		t.Logf("==> checking with flags %v", flagSet)
+		log.Printf("==> checking with flags %v", flagSet)
```

Comment:
Why this change? I don't think its needed?

Reply:
It’s needed because printing with t.Log causes the test case to fail on GitHub actions. Will share screenshots in a moment as I’m on mobile now.

This change doesn’t cause a change in behaviour but it does not cause a failure.

Reply:
![gherror](https://user-images.githubusercontent.com/25657923/69945331-5603be00-150f-11ea-9b15-6d1b70ca4aa3.png)

Please check this screenshot. You can see that the `t.Log` and `t.Logf` are treated as error and are causing the tests to fail. Hence the change.

Reply:
`t.Log` and `t.Logf` do not cause tests to fail, they just attach the logs to a particular test. [From the docs](https://golang.org/pkg/testing/#T.Log):

> For tests, the text will be printed only if the test fails or the -test.v flag is set.

I'm not sure where the `##[error]` formatting is coming from, but my guess is that we're just looking at the `stderr` stream and the test was failing because of something else further down. You can see that the error lines being printed still have a `PASS` underneath, although I'm not sure where to find that log in particular.

I've noticed that this project tends to print out a lot of things when any test fails, which is something we should address at some point, but `t.Logf` is almost always the right call in tests.

Reply:
@rliebz Please check

> It’s needed because printing with t.Log causes the test case to fail on GitHub actions.

I am aware that `t.Log` and `t.Logf` does not cause the test cases to fail. The GitHub actions environment is interpreting it as such. The tests are just outputting stuff to be verbose. They are not real errors.

I tried running the test cases on my machine and they ran fine. It's just on the GitHub Actions environment where they are failing. Hence this change.

Reply:
I couldn't find where the logs in the screenshot were coming from, so I had to play around with actions on my fork. I am not able to reproduce the behavior you're describing in this PR, although it appears we're only getting the ✅ on these because it's skipping the tests (see [my comment here](https://github.com/urfave/cli/pull/970/files#r354653289))

Whether or not we run `t.Logf` versus `log.Printf`, the tests are passing, but the error that's actually causing the test to fail is unrelated:

```
2019/12/06 03:36:42 exec: "gfmrun": executable file not found in $PATH
exit status 1
##[error]Process completed with exit code 1.
```

Since the `gfmrun` step only runs _after_ tests complete, I believes the unit tests are passing (albeit with a lot of extra noise), but the test step is failing because `gfmrun` is not available on the path. Turns out, we need to make it executable:

```diff
      - name: Install Dependencies
        run: |
          mkdir -p $GOPATH/bin
          curl -s https://api.github.com/repos/urfave/gfmrun/releases/latest | grep gfmrun-linux-amd64 | grep download | cut -d '"' -f 4 | wget -O $GOPATH/bin/gfmrun -i -
+         chmod +x $GOPATH/bin/gfmrun
          npm install markdown-toc
```

I then got even further, but ran into a strange error:
```
##[error]The operation was canceled.
```

Not sure what causes that. Because the symbol we're getting is a stop sign looking thing and not a big red X, I'm guessing the job is being terminated by GitHub actions for some reason--taking too long? Taking too much memory? Hard to say, Google didn't help me too much here. It passes sometimes and fails others when I re-run it. Note the symbols on the left:

<img width="720" alt="image" src="https://user-images.githubusercontent.com/5321575/70294553-e103e300-17b1-11ea-96ab-7e7f69664d4b.png">


The failures on macOS in that case are caused because we're still pulling in the linux version of `gfmrun`.

In terms of why logging could make it fail, from what I can tell, it's just bad UI on GitHub's part that gives misleading messages. GitHub actions is trying to be clever, and it considers anything that comes out of `stderr` to be an "error annotation". This is not really a good assumption to make, but it does not actually prevent the build from failing. In this case, the build passed with 10 error annotations:

<img width="1424" alt="image" src="https://user-images.githubusercontent.com/5321575/70294764-86b75200-17b2-11ea-88d6-f4d09b8188e4.png">

I realize that's a pretty big information dump, but hopefully that's enough context to help figure out what to do from here. If you're interested in digging deeper into the exact version of the code I ran or the logs, you should be able to see everything here: https://github.com/rliebz/cli/pull/1/checks?check_run_id=336127776

Reply:
I think I have managed to fix it. Thanks for all your work.